### DISCUSSION TITLE
Troubleshooting common problems

### DISCUSSION THREAD

**Discussion by Electrotechnique** [ElectroTechnique/electrotechnique.github.io#64] [Hardware Issues]
**First ensure you have the latest firmware.**

**Nothing works**
Check all solder joins. Check the Teensy is seated and making contact with the pin header. 
Check the SMD chips that came on the PCB are the correct way round and soldered (one PCB 
from the manufacturer had a MUX chip 180 degrees the wrong way!) 
**Really, 99% of problems have been caused by poor soldering**

**Display problems** 
If the display has inverted colours and is skewed, you probably have one that requires changes 
to the code in order to fix this. The solution is to modify the code that initialises the display (in 
ST7735Display.h). The following lines can be changed:
tft.initR(INITR_GREENTAB); //INITR_BLACKTAB 
 tft.invertDisplay(true); // false

Also [see this post](tinyurl.com/3m4nzyfp)
If the display rapidly changes, then you have a problem with the connection between the Teensy and the 
main PCB, which is causing noisy random parameter values because there is no connection to the pots. 
Make sure the Teensy pins are seated and making contact with the pin header or you have correctly 
soldered them.

**Audio problems**
A low speed clicking noise from the analogue audio jacks is probably due to using a mono plug in a 
stereo jack. Use stereo plugs with TSynth.
A low hum is usually a ground problem. Try a USB power pack or perhaps a USB isolator (a cheap, low 
speed one is fine) on the power cable. These sorts of problems are common with audio set-ups.
A faster clicking that varies with the volume setting is possibly the SCK on the display SPI line. This 
should no longer be such a problem with V1.3 PCBs. [See this post](tinyurl.com/y525f9ae)

There is a [Facebook TSynth builders group](https://tinyurl.com/ujusk4kx) if you want to talk to other builders.

**Comment by kfl02** [comment]
Quote: "If the display rapidly changes, then you have a problem with the connection between the Teensy and the
main PCB, which is causing noisy random parameter values because there is no connection to the pots.
Make sure the Teensy pins are seated and making contact with the pin header or you have correctly
soldered them."

After resoldering all of the pots three times I found out today that the 74hc4067 MUX1 chip is not correctly soldered to the PCB.

I always got random readings on Osc2 Pitch,, PWM Rate, Pitch Env Mod that connect to I13, I14 and I15 on MUX1. Everything else was working perfectly. 

Measuring the pots directly gave correct readings. Measuring at the corresponding pins at MUX1 did not. Measuring at the solder joints of the MUX1 were correct again.

So the MUX1 is not correctly soldered onto the PCB.

I have no experience with SMD soldering, but I will try to repair it tomorrow.

The PCB was ordered from Coq au Lo-Res in Germany.

**Reply by Electrotechnique** [reply]
Ok. There was one other instance of a MUX being soldered 180 degrees wrongly a couple of years back. Apparently JLCPCB's quality control doesn't extend to proper inspections. It should be easy to drag a soldering iron across the pins with some solder . Use flux if you have it and check with a magnifier for shorts between pins.

Any chance of a picture? I'm curious to see how it was poorly soldered.

**Reply by theinternal** [reply]
Hi, sorry you had to deal with this issue. Thank you for reporting this! I took the opportunity to check the remaining two PCBs we have in stock from the same manufacturing batch as yours and they are fine (fortunately).

I've read on FB that you were able to fix this in the meantime and have a functioning TSynth now, so congrats!
